src/stream/src/executor/backfill/cdc/utils.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use futures::pin_mut;
+use futures_async_stream::try_stream;
+use itertools::Itertools;
+use risingwave_common::array::{DataChunk, StreamChunk};
+use risingwave_common::catalog::{ColumnDesc, ColumnId, Schema};
+use risingwave_common::row::{Row, RowExt};
+use risingwave_common::types::{DataType, ScalarRefImpl};
+use risingwave_common::util::iter_util::ZipEqFast;
+use risingwave_connector::parser::{
+    DebeziumParser, EncodingProperties, JsonProperties, ProtocolProperties,
+    SourceStreamChunkBuilder, SpecificParserConfig,
+};
+use risingwave_connector::source::{SourceColumnDesc, SourceContext};
+
+use crate::executor::{BoxedMessageStream, Message, StreamExecutorError, StreamExecutorResult};
+
+#[try_stream(ok = Message, error = StreamExecutorError)]
+pub async fn transform_upstream(upstream: BoxedMessageStream, schema: &Schema) {
```

Comment:
`transform_upstream` seems only to be used in `cdc_backfill.rs`, so it doesn't need to be placed in such "utilities" mod. Actually putting it there make it easier to understand.

By the way, the related unit tests were still there. 🤣